src/GameLogic/PlugIns/GoldenInvasionPlugIn.cs

```diff
@@ -0,0 +1,390 @@
+                    }
+
+                    break;
+                }
+
+            case MobsInvasionState.Initialized:
+                {
+                    logger.LogInformation($"{state.MapName}: spawning mobs ...");
+                    state.NextRunUtc = DateTime.UtcNow.AddMilliseconds(configuration.EventDuration);
+                    state.State = MobsInvasionState.Started;
+
+                    await this.SpawnGeneralMobsAsync(state).ConfigureAwait(false);
+                    await this.SpawnUniqueMobsAsync(state).ConfigureAwait(false);
+
+                    logger.LogInformation($"{state.MapName}: spawning finished");
+
+                    break;
+                }
+
+            case MobsInvasionState.Started:
+                {
+                    logger.LogInformation($"{state.MapName}: event finished");
+
+                    state.State = MobsInvasionState.NotStarted;
+
+                    try
+                    {
+                        await gameContext.ForEachPlayerAsync(p => this.TrySendFlyingDragonsAsync(p, false)).ConfigureAwait(false);
+                    }
+                    catch (Exception ex)
+                    {
+                        Debug.Fail(ex.Message, ex.StackTrace);
+                    }
+
+                    break;
+                }
+        }
+    }
+
+    /// <inheritdoc />
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "VSTHRD100:Avoid async void methods", Justification = "Catching all Exceptions.")]
+    public async void ObjectAddedToMap(GameMap map, ILocateable addedObject)
+    {
+        try
+        {
+            if (addedObject is Player player)
+            {
+                var state = GetStateByGameContext(player.GameContext);
+
+                var flyingEnabled = state.State != MobsInvasionState.NotStarted;
+
+                await this.TrySendFlyingDragonsAsync(player, flyingEnabled).ConfigureAwait(false);
+            }
+        }
+        catch
+        {
+            // must be catched because it's an async void method.
+        }
+    }
+
+    private static GameServerState GetStateByGameContext(IGameContext gameContext)
+    {
+        return _states.GetOrAdd(gameContext, gameCtx => new GameServerState(gameContext));
+    }
+
+    private async Task SpawnUniqueMobsAsync(GameServerState state)
+    {
+        var gameContext = state.Context;
+
+        foreach (var (mapId, mobs) in UniqueMobsCountByMapId)
+        {
+            var gameMap = await gameContext.GetMapAsync(mapId).ConfigureAwait(false);
+
+            if (gameMap is null)
+            {
+                continue;
+            }
+
+            foreach (var (mobId, mobsCount) in mobs)
+            {
+                var monsterDefinition = gameContext.Configuration.Monsters.First(m => m.Number == mobId);
+
+                var tasks = Enumerable.Repeat(() => CreateMonsterAsync(gameContext, gameMap, monsterDefinition, 10, 240, 10, 240), mobsCount).Select(c => c());
```

Comment:
Any other way?

Reply:
You can pass mobsCount to CreateMonsterAsync, and set it as quantity.
Then I wouldn't start a task per monster, but just await each call inside the loop. Was that maybe the reason for your concurrency errors? ;-)